### PR TITLE
Validation component for MP OpenAPI feature.

### PR DESCRIPTION
**Pull request by mrglavas**
Signed-off-by: Michael Glavassevich <mrglavas@ca.ibm.com>

### CODE REVIEW

**Comment by mrglavas** [comment]
@OpenLiberty/message-reviewer The messages in dev/com.ibm.ws.microprofile.openapi/resources/com/ibm/ws/microprofile/openapi/impl/validation/resources/ValidationMessages.nlsprops are identical to the messages contained in the file with the same name included in the openapi-3.0 feature.

**Comment by mrglavas** [comment]
#build

File: dev/com.ibm.ws.microprofile.openapi/resources/com/ibm/ws/microprofile/openapi/impl/validation/resources/ValidationMessages.nlsprops [hunk 1]
```diff
@@ -0,0 +1,89 @@
...
+pathItemOperationParameterNotDeclaredSingle=The PathItem object must contain a valid path. The \"{0}\" operation from the \"{1}\" path defines one path parameter that is not declared: \"{2}\".
+pathItemOperationParameterNotDeclaredMultiple=The PathItem object must contain a valid path. The \"{0}\" operation from the \"{1}\" path defines \"{2}\" path parameters that are not declared: \"{3}\".
+pathItemOperationNoPathParameterDeclared=The PathItem object must contain a valid path. The \"{0}\" operation of the \"{1}\" path does not define a path parameter that is declared: \"{2}\".
+pathItemInvalidFormat=The PathItem object must contain a valid path. The format of the \"{0}\" path is invalid.
+
+pathsRequiresSlash=The Paths object must contain a valid path. The \"{0}\" path does not begin with a slash.
+
+referenceExternalOrExtension=The \"{0}\" value is an external reference or an extension. No validation available.
+referenceNotPartOfModel=The specified reference \"{0}\" is not part of the model components.
+referenceNotValid=The specified reference \"{0}\" is not valid.
```
Comment (bsbyrd1) [inline dev/com.ibm.ws.microprofile.openapi/resources/com/ibm/ws/microprofile/openapi/impl/validation/resources/ValidationMessages.nlsprops]: The specified \"{0}\" reference is not valid.

File: dev/com.ibm.ws.microprofile.openapi/resources/com/ibm/ws/microprofile/openapi/impl/validation/resources/ValidationMessages.nlsprops [hunk 2]
```diff
@@ -0,0 +1,89 @@
...
+referenceNotValidFormat=The \"{0}\" reference is not in a valid format.
+referenceNull=The \"{0}\" model components object is null or $ref \"{1}\" is null.
+referenceToObjectInvalid=The \"{0}\" value is an invalid reference for \"{1}\".
```
Comment (bsbyrd1) [inline dev/com.ibm.ws.microprofile.openapi/resources/com/ibm/ws/microprofile/openapi/impl/validation/resources/ValidationMessages.nlsprops]: Specify the object type for \"{1}\".

File: dev/com.ibm.ws.microprofile.openapi/resources/com/ibm/ws/microprofile/openapi/impl/validation/resources/ValidationMessages.nlsprops [hunk 3]
```diff
@@ -0,0 +1,89 @@
...
+
+responseMustContainOneCode=The Responses object must contain at least one response code.
+responseShouldContainSuccess=The Responses object should contain at least one response code for a successful operation.
+
+securityRequirementNotDeclared=The \"{0}\" name provided for the SecurityRequirement object does not correspond to a declared security scheme.
+securityRequirementScopeNamesRequired=The SecurityRequirement object''s \"{0}\" field should be a list of scope names required for execution, but is: \"{1}\".
```
Comment (bsbyrd1) [inline dev/com.ibm.ws.microprofile.openapi/resources/com/ibm/ws/microprofile/openapi/impl/validation/resources/ValidationMessages.nlsprops]: The \"{0}\" field for the SecurityRequirement object should be a list of scope names that are required for execution.

File: dev/com.ibm.ws.microprofile.openapi/resources/com/ibm/ws/microprofile/openapi/impl/validation/resources/ValidationMessages.nlsprops [hunk 4]
```diff
@@ -0,0 +1,89 @@
...
+securityRequirementFieldNotEmpty=The SecurityRequirement object''s \"{0}\" field should be empty, but is: \"{1}\".
```
Comment (bsbyrd1) [inline dev/com.ibm.ws.microprofile.openapi/resources/com/ibm/ws/microprofile/openapi/impl/validation/resources/ValidationMessages.nlsprops]: The \"{0}\" field  for the SecurityRequirement object should be empty.

File: dev/com.ibm.ws.microprofile.openapi/resources/com/ibm/ws/microprofile/openapi/impl/validation/resources/ValidationMessages.nlsprops [hunk 5]
```diff
@@ -0,0 +1,89 @@
...
+securityRequirementIsEmpty=The SecurityRequirement object must not be empty.
+
+securitySchemeInFieldInvalid=The \"{0}\" SecurityScheme object has an error. The value of its \"in\" field is \"{1}\", but must be one of (\"query\", \"header\", \"cookie\").
```
Comment (bsbyrd1) [inline dev/com.ibm.ws.microprofile.openapi/resources/com/ibm/ws/microprofile/openapi/impl/validation/resources/ValidationMessages.nlsprops]: The \"{0}\" SecurityScheme object has an error. The value of its \"in\" field is \"{1}\". However, you must specify one of following values: (\"query\", \"header\", \"cookie\")

File: dev/com.ibm.ws.microprofile.openapi/resources/com/ibm/ws/microprofile/openapi/impl/validation/resources/ValidationMessages.nlsprops [hunk 6]
```diff
@@ -0,0 +1,89 @@
...
+securitySchemeInvalidURL=The SecurityScheme object must contain a valid URL. The \"{0}\" value specified for the URL is not valid.
+securitySchemeNonApplicableField=One or more fields defined are not applicable to a SecurityScheme instance of type \"{0}\".
+
+serverVariableNotDefined=The \"{0}\" variable in the Server object is not defined.
+serverInvalidURL=The Server object must contain a valid URL. The \"{0}\" value specified for the URL is not valid.
+
+requiredFieldMissing=Validation error at location: \"{0}\". Required \"{1}\" field is missing or is set to an invalid value.
```
Comment (bsbyrd1) [inline dev/com.ibm.ws.microprofile.openapi/resources/com/ibm/ws/microprofile/openapi/impl/validation/resources/ValidationMessages.nlsprops]: A validation error occurred at the following location: \"{0}\". The required \"{1}\" field is missing or is set to an invalid value.

**Comment by mrglavas** [comment]
@bsbyrd1: Wondering why some of the replacement parameters are missing in your suggested changes. For example: "Specify the object type for "{1}"." and "The "{0}" field for the SecurityRequirement object should be a list of scope names that are required for execution."

Also, these messages exist verbatim in Commercial Liberty. The purpose of this work item was just to import them here (along with the rest of the validator component). Planning to open a new work item with another pull request to make message improvements separate from this pull request. I will link it to this PR.

**Comment by mrglavas** [comment]
#build

**Comment by mrglavas** [comment]
Not merging into integration at this time.

**Pull request closed without merging**